### ISSUE TITLE
Can't read gamestate

### ISSUE THREAD

**Issue by andy17110** [Nro001/Nro-Galaxy-Editor#6]
Something happned in patch 3.10 and It can't read new gamestate files. Still works with older saves, not the new ones.

**Comment by fircrag** [comment]
I also faced this problem

**Comment by FranciscoManuelDiasP** [comment]
Same here, the log just says:
`Godot Engine v3.5.1.stable.official.6fed1ffa3 - https://godotengine.org
OpenGL ES 2.0 Renderer: GeForce MX130/PCIe/SSE2
 
「Nro's Galaxy Editor」
「Directory」 Exist----------------------------< user://galactic_object/
「Directory」 Exist----------------------------< user://packed/
「Directory」 Exist----------------------------< user://unpacked/
「Directory」 Exist----------------------------< user://parsed/
「Directory」 Cleared--------------------------< user://parsed/
「File」 Read----------------------------------< gamestate
「Parse」 Trimming-----------------------------< user://unpacked/gamestate
「Directory」 Cleared--------------------------< user://galactic_object/
「Parse」 Partitioning-------------------------< user://unpacked/galactic_object
ERROR: Can't take value from empty array.
   at: back (core/array.cpp:160) - Condition "_p->array.size() == 0" is true. Returned: Variant()`

**Comment by FrancisMDP** [comment]
Me too.
The log says:
```
Godot Engine v3.5.1.stable.official.6fed1ffa3 - https://godotengine.org
OpenGL ES 2.0 Renderer: GeForce MX130/PCIe/SSE2
 
「Nro's Galaxy Editor」
「Directory」 Exist----------------------------< user://galactic_object/
「Directory」 Exist----------------------------< user://packed/
「Directory」 Exist----------------------------< user://unpacked/
「Directory」 Exist----------------------------< user://parsed/
「Directory」 Cleared--------------------------< user://parsed/
「File」 Read----------------------------------< gamestate
「Parse」 Trimming-----------------------------< user://unpacked/gamestate
「Directory」 Cleared--------------------------< user://galactic_object/
「Parse」 Partitioning-------------------------< user://unpacked/galactic_object
ERROR: Can't take value from empty array.
   at: back (core/array.cpp:160) - Condition "_p->array.size() == 0" is true. Returned: Variant()
```

**Comment by DarcyWenn** [comment]
Also having this issue

**Comment by JohnnyGAB1108** [comment]
The issue is that the 3.7 “Canis Minor” update changed the scripting structure of the save files from something like

galactic_object={
      (star ID)={
      }
}

to

galactic_object=
{
      (star ID)=
      {

      }
}

So the the galaxy editor program is looking for the for the old format and needs to be retooled to recognize the new format.

**Comment by MCGorbachev** [comment]
Confirm, the app crashes immediately after dropping gamestate file

**Comment by WillaGrey** [comment]
The same happens to me.

**Comment by PadOpinion** [comment]
As a workaround, replace regex `=\n\s*{` with `={`. The output doesn't need to be fixed, Stellaris car read both formats.

**Comment by JohnnyGAB1108** [comment]
> As a workaround, replace regex `=\n\s*{` with `={`. The output doesn't need to be fixed, Stellaris car read both formats.

How would one do that in Notepad+?

**Comment by WillaGrey** [comment]
Go to Replace mode (next to search) and check the Regular Expression tick in the bottom left corner.

**Comment by JohnnyGAB1108** [comment]
> Go to Replace mode (next to search) and check the Regular Expression tick in the bottom left corner.

Nice
Thank you

**Comment by Nro001** [comment]
I fixed this problem with v1.6 now, thanks for helping others @PadOpinion! 

#7
